extensions/identity-hub-verifier/src/main/java/org/eclipse/dataspaceconnector/identityhub/verifier/IdentityHubCredentialsVerifier.java

```diff
@@ -61,41 +68,74 @@ public IdentityHubCredentialsVerifier(IdentityHubClient identityHubClient, Monit
+            monitor.severe(errorMessage);
+            return Result.failure(errorMessage);
         }
 
-        var jwts = identityHubClient.getVerifiableCredentials(hubBaseUrl.getContent());
-        if (jwts.failed()) {
-            return Result.failure(jwts.getFailureMessages());
+        monitor.debug(() -> String.format("Using identity hub URL: %s", hubBaseUrl.get()));
+
+        var verifiableCredentials = identityHubClient.getVerifiableCredentials(hubBaseUrl.get());
+        if (verifiableCredentials.failed()) {
+            monitor.severe("Could not retrieve verifiable credentials from identity hub");
+            return Result.failure(verifiableCredentials.getFailureMessages());
         }
-        var verifiedJwt = jwts.getContent()
+
+        monitor.debug(() -> String.format("Retrieved %s verifiable credentials from identity hub", verifiableCredentials.getContent().size()));
+
+        var verifiedCredentials = verifyCredentials(verifiableCredentials, didDocument);
+
+        monitor.debug(() -> String.format("Verified %s credentials", verifiedCredentials.size()));
+
+        var claims = extractClaimsFromCredential(verifiedCredentials);
+
+        return Result.success(claims);
+    }
+
+    @NotNull
+    private List<SignedJWT> verifyCredentials(StatusResult<Collection<SignedJWT>> jwts, DidDocument didDocument) {
+        var result = jwts.getContent()
                 .stream()
-                .filter(jwt -> jwtCredentialsVerifier.verifyClaims(jwt, didDocument.getId()))
-                .filter(jwtCredentialsVerifier::isSignedByIssuer);
+                .collect(partitioningBy((jwt) -> jwtCredentialsVerifier.verifyClaims(jwt, didDocument.getId()) && jwtCredentialsVerifier.isSignedByIssuer(jwt)));
 
-        var partitionedResult = verifiedJwt.map(verifiableCredentialsJwtService::extractCredential).collect(partitioningBy(AbstractResult::succeeded));
-        var successfulResults = partitionedResult.get(true);
-        var failedResults = partitionedResult.get(false);
+        var successfulResults = result.get(true);
+        var failedResults = result.get(false);
 
-        failedResults.forEach(result -> monitor.warning(String.join(",", result.getFailureMessages())));
+        if (!failedResults.isEmpty()) {
+            monitor.warning(String.format("Ignoring %s invalid verifiable credentials", failedResults.size()));
+        }
+
+        return successfulResults;
+    }
+
+    @NotNull
+    private Map<String, Object> extractClaimsFromCredential(List<SignedJWT> verifiedCredentials) {
+        var result = verifiedCredentials.stream()
+                .map(verifiableCredentialsJwtService::extractCredential)
+                .collect(partitioningBy(AbstractResult::succeeded));
+
+        var successfulResults = result.get(true);
+        var failedResults = result.get(false);
 
-        var claims = successfulResults.stream()
+        if (!failedResults.isEmpty()) {
```

Comment:
This is dangerous: what if a dataspace requires that all claims must be verified? Maybe you could introduce a `VerificationResult` as a subclass of `AbstractResult`, that allows for transporting both the successful claims, and the failed ones? That way, the caller of the `verifyCredentials` method could decide.

Reply:
--> will be done in another issue/PR

Reply:
As discussed, this will be tackled in a separate story: https://github.com/agera-edc/IdentityHub/issues/47. This behaviour was not changed with the current PR.